### ISSUE TITLE
Add model for Bayesian Committee Machine

### ISSUE THREAD

**Issue by john-waczak** [john-waczak/MLJGP.jl#5]
Due to the need to invert the kernel matrix `K`, fitting a GPR model scales as $\mathcal{O}(n^3)$. There are a variety of strategies to get around this including the [Bayesian Committee Machine](https://www.dbs.ifi.lmu.de/~tresp/papers/bcm6.pdf) which splits the training set up into random subsets of size `M<N` with individual GPs trained on each. 

We could easily implement this, and since hyperparameters are shared between each of the committee members, the gradient calculation step can be parallelized. See [this python version](https://github.com/jopago/GPyBCM) for implementation ideas.
